File: src/solar/solar-common/decorator/OnPlayerChat.ts
export default function OnPlayerChat(chatMessage: string) {
    return function (method: (...args) => any, context: ClassMethodDecoratorContext) {
        se.onPlayerChat(chatMessage, () => {
            print("onPlayerChat测试:" + tostring(context.name))
            method.apply(null);
        });
        return method;
    };
}

File: scripts/solar/solar-common/decorator/OnPlayerChat.lua
local ____lualib = require("lualib_bundle")
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["5"] = 1,["6"] = 2,["7"] = 3,["8"] = 3,["9"] = 3,["10"] = 4,["11"] = 5,["12"] = 3,["13"] = 3,["14"] = 7,["15"] = 2});
local ____exports = {}
function ____exports.default(self, chatMessage)
    return function(self, method, context)
        se:onPlayerChat(
            chatMessage,
            function()
                print("onPlayerChat测试:" .. tostring(context.name))
                method(nil)
            end
        )
        return method
    end
end
return ____exports
